tests/appsec/iast/utils.py

```diff
@@ -234,24 +239,43 @@ def validate_stack_traces(request):
     assert "frames" in stack_trace, "'frames' not found in stack trace"
     assert len(stack_trace["frames"]) <= 32, "stack trace above size limit (32 frames)"
 
-    # Vulns without location path are not expected to have a stack trace
+    # Vulns without location path/class are not expected to have a stack trace
+    old_version = context.library < "java@1.47.0"
     location = vuln["location"]
-    assert location is not None and "path" in location, "This vulnerability is not expected to have a stack trace"
+    if old_version:
+        assert location is not None and "path" in location, "This vulnerability is not expected to have a stack trace"
```

Comment:
To make assertions clearer (also when they fail), can you move `assert location, "No location found"` above the conditional?

Then, while this was written before this PR, the assertion message here (`"This vulnerability is not expected to have a stack trace"`) seems to be onrelated. Should be something like `Expected the vulnerability location to have a path`.